bokehjs/src/coffee/models/transforms/interpolator.coffee

```diff
@@ -0,0 +1,85 @@
+_ = require "underscore"
+Transform = require "./transform"
+p = require "../../core/properties"
+
+class Interpolator extends Transform.Model
+
+  initialize: (attrs, options) ->
+    super(attrs, options)
+    @_x_sorted = []
+    @_y_sorted = []
+    @_sorted_dirty = true
+
+    @bind 'change', () ->
+      console.log 'Changed the configuration'
+      @_sorted_dirty = true
+
+  props: ->
+    return _.extend {}, super(), {
+      x: [ p.Any, '']
+      y: [ p.Any, '']
+      data: [ p.Any, null]
+    }
+
+  sort: (descending = false) ->
+    # Verify that all necessary objects exist...
+    if typeof(@get('x')) != typeof(@get('y'))
+      # ToDo: Throw a resonable error here
+      return
+
+    if typeof(@get('x')) == 'object'
+      if @get('x').length != @get('y').length
+        # ToDo: Throw a resonable error here
+        return
+    else
+      if @get('x').length == 0
+        return
+
+      if @get('y').length == 0
+        return
+
+      if @get('data') == null
+        # ToDo: Throw a resonable error here.
+        return
+
+    # Stop processing this if the dirty flag is not set
+    if(@_sorted_dirty == false)
+      return
+
+    tsx = []
+    tsy = []
+
+    # Populate the tsx and tsy variables correctly depending on the method by which the user populated the interpolation
+    # data.
+    if typeof(@get('x')) == 'string'
+      data = @get('data')
+
+      tsx = data.get_column(@get('x'))
+      tsy = data.get_column(@get('y'))
+    else
+      tsx = @get('x')
+      tsy = @get('y')
```

Comment:
I think some of this logic could be simplified a bit with a slight refactoring along these lines:

```
# _get_values has the logic to check for literal sequence, or column in source
tsx = @_get_values(@get('x'))
tsy = @_get_values(@get('y'))

if tsx.length != tsy.length
  # raise error

if tsx.length == 0 
  return
```